### PR TITLE
Use rust-g for revision info instead of parsing .git

### PR DESCRIPTION
**Pull request by SpaceManiac**
:cl:
code: Git revision info is now gathered by rust-g instead of manually reading .git files.
/:cl:

Following https://github.com/tgstation/rust-g/pull/9.

* Slightly tweaks code and formatting of revision info output in logs and verb
* Puts revision info in public runtime logs, so runtime errors can be more easily correlated to git info

**Needs work between approval and merging**:
* [x] Cutting a rust-g release
* [x] Bumping `RUST_G_VERSION`
* [x] Committing updated `rust_g.dll` to the repo

### CODE REVIEW

**Comment by JJRcop** [comment]
Why not make work a checklist.

Checklists are:
- [x] cool
- [x] show up on pr summaries
- [ ] hard to use

```markdown
Checklists are:
- [x] cool
- [x] show up on pr summaries
- [ ] hard to use
```

**Comment by MrStonedOne** [comment]
@Cyberboss will this work with how tgs3 does the things.

**Comment by Cyberboss** [comment]
Yes and no, I recently added code to short circuit our .git parsing if TGS is present since it delivers more info via the API. But the way V3 code does it is the same as old getrev.

TL;DR will work fine for us, new V3 users may have a slight runtime till I fix it

File: code/datums/helper_datums/getrev.dm [hunk 1]
```diff
@@ -1,44 +1,40 @@
 /datum/getrev
-	var/originmastercommit
-	var/commit
-	var/list/testmerge = list()
+	var/commit  // git rev-parse HEAD
 	var/date
+	var/originmastercommit  // git rev-parse origin/master
```
Comment (Cyberboss) [inline code/datums/helper_datums/getrev.dm]: `origin/<current branch>` is more accurate, but w/e

Comment (Cyberboss) [reply code/datums/helper_datums/getrev.dm]: I suppose this is a difference in how TGS operates actually because tg shitrev code only tracks origin/master bleh

**Comment by Cyberboss** [comment]
Just the .git/logs directory. But the rust-g stuff won't run at all under
TGS due to the short circuit at the start of getrev/New

File: code/datums/helper_datums/getrev.dm (hunk 1 above)
Comment (SpaceManiac) [reply code/datums/helper_datums/getrev.dm]: it's possible in git to learn the remote + branch corresponding to a local branch (dmdoc does this) but it's kind of a PITA so I didn't want to include it for now

**Comment by SpaceManiac** [comment]
After this PR probably going to let it pull revision information from a file and use that instead of copying `.git` in the docker scripts and such as well

**Comment by SpaceManiac** [comment]
rust-g release cut, passing travis. Would appreciate it if someone double-checked that the Windows binary works.

**Comment by optimumtact** [comment]
is there any point in test merging this given that TGS is short circuiting it anyway?

**Review by Cyberboss — APPROVED** [review]

**Comment by MrStonedOne** [comment]
revision parsing is still broken, tgs3 is not short circuiting anything, commit hash hasn't been working since this pr was merged.

**Comment by Cyberboss** [comment]
That's odd, it should be using the old method here: https://github.com/tgstation/tgstation/blob/master/code/modules/tgs/v3210/api.dm#L67

Ohh you know what it is, getrev is initializing before TGS

**Comment by Cyberboss** [comment]
[Wait no...](https://github.com/tgstation/tgstation/blob/master/code/game/world.dm#L19)
